src/library_html5.js

```diff
@@ -314,7 +316,11 @@ var LibraryHTML5 = {
     // values we accept here, EMSCRIPTEN_EVENT_TARGET_* (which map to 0, 1, 2).
     // In other words, if cString > 2 then it's a pointer to a valid place in
     // memory, and points to a C string.
+#if MEMORY64
+    return cString > 2 ? UTF8ToString(Number(cString)) : cString;
+#else
     return cString > 2 ? UTF8ToString(cString) : cString;
```

Comment:
My understanding is that `cString` here should already be in number format .. since it should have been converted to a number as soon as the left the native side (due to the automatic wrapper function generated by the __sig attributes).  

Can you find the callstack that leads to this not being a number?